src/docfx/lib/git/GitUtility.cs

```diff
@@ -21,6 +22,8 @@ namespace Microsoft.Docs
     /// </summary>
     public static class GitUtility
```

Comment:
Consider split the native GetCommits implementation into a separate file, using partial class.

Reply:
no.. please do not use partial class....

---
In reply to: [180358099](https://github.com/dotnet/docfx/pull/2611#discussion_r180358099) [](ancestors = 180358099)

Reply:
i think it's worth to do that, as the native code git mix gitbas make the current git utility complex

---
In reply to: [180613821](https://github.com/dotnet/docfx/pull/2611#discussion_r180613821) [](ancestors = 180613821,180358099)

Reply:
Split is good, but don't use partial class:blush: 

---
In reply to: [180615520](https://github.com/dotnet/docfx/pull/2611#discussion_r180615520) [](ancestors = 180615520,180613821,180358099)

Reply:
why not?

---
In reply to: [180616171](https://github.com/dotnet/docfx/pull/2611#discussion_r180616171) [](ancestors = 180616171,180615520,180613821,180358099)

Reply:
[https://stackoverflow.com/questions/3601901/when-is-it-appropriate-to-use-c-sharp-partial-classes](https://stackoverflow.com/questions/3601901/when-is-it-appropriate-to-use-c-sharp-partial-classes)

---
In reply to: [180616334](https://github.com/dotnet/docfx/pull/2611#discussion_r180616334) [](ancestors = 180616334,180616171,180615520,180613821,180358099)